### PR TITLE
Prevent expensive JOINs when there aren't multiple currencies used

### PR DESCRIPTION
**Pull request by joshbetz**
#### Changes proposed in this Pull Request

Prevent expensive JOINs when there aren't multiple currencies used

Looks up the number of currencies that have been used on this site. If
there are more than 1, we need to reflect that in the analytics.
Otherwise we should avoid the expensive JOINs.

See #4280

#### Testing instructions (updated by @dmallory42)

Checking the MC Analytics work when a MC order has been made

- On a site where you have previously made a multi-currency order, go to the analytics section and verify (either using step debugging, or query monitor to view the queries that are being run) to ensure that when the `Analytics` code is initialised, it does *not* return at [this point](https://github.com/Automattic/woocommerce-payments/blob/a4b6b377ab64fe0e389f74bfbfd142b7193a3979/includes/multi-currency/Analytics.php#L72). 
- Make an order with a multi-currency, and view the analytics for today to ensure that the order appears, and the figures (converted into your store default currency) look sensible.

Checking Analytics still loads when no MC order has been made

- Edit the `has_multi_currency_orders` function to return false, then check again - this time , the initialisation of MC Analytics should return early, before adding the filters. Check the analytics and verify that nothing is broken, all pages load without errors etc. (note that the total figures *may* be incorrect in this case, because the filters were not added and the store has MC orders, in prod this should not happen since we'd only return early if there are no MC orders).

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by joshbetz** [comment]
@dmallory42 I think this will miss a big use-case I was thinking of where multi-currency is enabled, but only 1 currency has ever been used. I suspect this is true for many stores--especially any stores that primarily do in person payments or local delivery. It seems like the multi-currency support is not useful in that case?

**Comment by dmallory42** [comment]
>  I think this will miss a big use-case I was thinking of where multi-currency is enabled, but only 1 currency has ever been used. I suspect this is true for many stores--especially any stores that primarily do in person payments or local delivery. It seems like the multi-currency support is not useful in that case?

Hi! That use case should be covered by this implementation. 🙂 In the case that orders have only ever been placed in the store default currency, the metadata key `_wcpay_multi_currency_order_exchange_rate` will not be present, therefore `has_multi_currency_orders` should return `false` and the JOINs will not be added.

File: includes/multi-currency/Analytics.php [hunk 1]
```diff
@@ -332,6 +333,32 @@ private function generate_case_when( string $variable, string $then, string $els
...
+		$orders = intval(
+			$wpdb->get_var(
+				"
+				SELECT
+					COUNT(DISTINCT orders.ID)
+				FROM
+					{$wpdb->posts} AS orders
+				INNER JOIN
+					{$wpdb->postmeta} order_meta ON order_meta.post_id = orders.ID
+					AND order_meta.meta_key = '_wcpay_multi_currency_order_exchange_rate';
```
Comment (ismaeldcom) [inline includes/multi-currency/Analytics.php:354]: Not a big improvement, but thinking on squeezing every ms, maybe we could avoid the `JOIN` and just look for the meta. In the end we only want to know if there are orders with exchange rate.

```suggestion
				SELECT
					COUNT(post_id)
				FROM
					{$wpdb->postmeta}
				WHERE
					meta_key = '_wcpay_multi_currency_order_exchange_rate';
```

**Review by ismaeldcom — APPROVED** [review]
Thanks @joshbetz  and @dmallory42 for working on this, it's a big performance improvement for those sites using only the store currency while having Multi-Currency enabled.

Looks good to me! Just a tiny check to prevent the performance hit when possible. I've only added a nitpick, but feel free to merge without it 🙂

File: includes/multi-currency/Analytics.php (hunk 1 above)
Comment (dmallory42) [reply includes/multi-currency/Analytics.php:354]: Good idea! That would achieve the same result in a more efficient way. Committed the suggestion!

**Comment by joshbetz** [comment]
Awesome. Looks good to me.

**Pull request merged**